src/server.c

```diff
@@ -596,6 +600,7 @@ int dqlite_node_start(dqlite_node *t)
 	}
 
 	if (!taskReady(t)) {
+		dqlite_node_stop(t);
```

Comment:
Why do we need this call? As far as I can tell `taskReady()` can only return `false` if `taskRun` calls `sem_post` and in that case it will return and make the thread exit. In that case the UV loop won't be running and the call to `uv_async_send` inside `dqlite_node_stop` will fail.

Maybe instead of calling `dqlite_node_stop` you could simply call `pthread_join`, although it shouldn't be strictly necessary.

Reply:
I can drop this commit for now so we're not blocking other stuff in here which might be valuable.